### ISSUE TITLE
Authentification keycloak apimanager case sensitive

### ISSUE THREAD

**Issue by GFlo69** [apiman/apiman#1413]
Hi,

On apiman version 2.1.0, the application of user rights is case-sensitive.
On version 2.0.0, this is not the case

Bests regards

**Comment by msavy** [comment]
Thanks for the report, @GFlo69 

Can you provide a bit more information, please? I'm not sure what you mean by 'user rights'. Can you provide an explicit example of your issue.

What is your configuration?

**Comment by GFlo69** [comment]
My case is a bit specific. I am in a migration phase.

When I load my export in the new version of apiman with the users in uppercase, authentication via an external keycloak returns me a user in lowercase.

Authentication and access to apimanui is not a problem. However, the management of access to organizations is not applied correctly.

In my case I end up with two identical users with an upper / lower case

**Comment by msavy** [comment]
What is your Apiman Manager API datastore? Is it a SQL database or Elasticsearch?

**Comment by GFlo69** [comment]
Elasticsearch V7.5

**Comment by volkflo** [comment]
Which version of keycloak are you using?

**Comment by GFlo69** [comment]
Hi,

Keycloak : 12.0.4
I cannot update keycloak.

Also, I don't think my problem is with keycloak.

I do have a behavior gap between version 2.0.0 and 2.1.0 on the same keycloak.

Without knowing the apiman code, I will tend to say that the prefered_username provided by the keycloak adapter should be "uppersized" by apiman, right?

**Comment by volkflo** [comment]
I am not totally sure how you got to this point (maybe this is caused due to some mapping changes).
But keycloak stores the username in lowercase almost since the beginning https://issues.redhat.com/browse/KEYCLOAK-2858.
So I am not sure how the usernames in uppercase got created.
I would suggest that you search and replace the usernames in the export with the lowercase ones and then import the data

**Comment by GFlo69** [comment]
"I suggest you find and replace usernames in export with lowercase, then import the data" ==> Yes that's what I'm realizing. But this gap remains strange between the two versions 2.0 and 2.1

**Comment by msavy** [comment]
> "I suggest you find and replace usernames in export with lowercase, then import the data" ==> Yes that's what I'm realizing. But this gap remains strange between the two versions 2.0 and 2.1

I try my best, but sometimes bugs and other issues happen when making a new release. 

Unfortunately, I likely don't have time to fully investigate today.

**Comment by volkflo** [comment]
That's true, we will definitely keep this issue open and try to investigate when we have time.
If you find something on your own, it would be great if you could share that with us :)

**Comment by msavy** [comment]
BTW, this will likely be helpful: `jq your_jq_query |= ascii_downcase' file`
